tests/loom.rs

```diff
@@ -0,0 +1,36 @@
+#[cfg(loom)]
+#[cfg(test)]
+mod loom_tests {
+    extern crate evmap;
+
+    use loom::sync::atomic::AtomicUsize;
+    use loom::thread;
+
+    use std::sync::atomic::Ordering::SeqCst;
+    use std::sync::Arc;
+
+    #[test]
+    fn evmap_read_while_remove() {
+        loom::model(|| {
+            let (r, mut w) = evmap::new();
+            w.insert(1, 2);
+            w.refresh();
+
+            let val = Arc::new(AtomicUsize::new(0));
+            let val_copy = Arc::clone(&val);
+
+            let read_thread = thread::spawn(move || {
+                val.store(*r.get_one(&1).as_deref().unwrap(), SeqCst);
+            });
+
+            let write_thread = thread::spawn(move || {
+                w.remove_entry(1);
```

Comment:
You don't need to spawn a thread for performing this operation. You can just perform it directly on the main thread, and loom will still handle scheduling the different interleavings.

Reply:
Oh so this is something I was messing around with yesterday when I was trying to figure out what it was doing. If I don't spawn a thread here the test just passes instantly.

Reply:
Also, and I don't know if I'm doing this right, if I remove this thread and then put an `eprintln` in `remove_entry` and `get_one` I only see this output in the test:
```
running 1 test
Removing entry!
Reading value!
Completed in 1 iterations
test loom_tests::evmap_read_while_remove ... ok
```
Which makes me think `loom` never tried the version where we read the value before removing the entry? Am I interpreting that correctly?

Reply:
Unless `loom` determined through it's state-squashing magic that there was no need to reorder them 🤔

Reply:
Ah, yes, this is because there's no synchronization happening on the write path here! Basically, the call to `remove_entry` just adds a remove to the operation log, and does nothing else, so loom (correctly) determines that there is no need to try another schedule. Running the other thread cannot possibly affect the execution of the remover thread. If you had a refresh in there on the other hand, it would. You could force it by using `thread::yield_now`, but basically loom eliminating this state is sufficient proof. If we _did_ add synchronization to `remove_entry` later, then loom _would_ run that additional execution and tell us if there was a mistake.

Reply:
That makes sense 👍 

Follow up question: Is it worth writing `loom` tests for things where there are no "extra" scheduling options? Currently this is doing "nothing special" on top of the normal test suite. Do we want this here so that, if `remove_entry` _gains_ some synchronization code, we're "ready" for it?

I'm torn because I could see this catching an error if synchronization is added but I'm generally against writing code for futures which may never exist. This isn't my library and there may be some solid value there but my initial gut reaction is that the loom tests should be added when the synchronization logic is added. And this PR would only be adding those `loom` tests that actually exercise various thread sync logic.

What do you think?

Reply:
My cautious vote would be to keep it. These things are subtle enough that guarding towards accidental future changes are worthwhile. Especially since this test takes so little time to run (as opposed to most loom tests). I would make sure to include a comment saying why it's there given its perceive lack of effect though.